base/multidimensional.jl

```diff
@@ -690,14 +687,15 @@ julia> diff(vec(a))
  12
 ```
 """
-function diff(A::AbstractMatrix; dims::Integer)
-    if dims == 1
-        [A[i+1,j] - A[i,j] for i=1:size(A,1)-1, j=1:size(A,2)]
-    elseif dims == 2
-        [A[i,j+1] - A[i,j] for i=1:size(A,1), j=1:size(A,2)-1]
-    else
-        throw(ArgumentError("dimension must be 1 or 2, got $dims"))
-    end
+function diff(a::AbstractArray; dims::Integer)
+  @assert !has_offset_axes(a)
```

Comment:
This should use the Julia standard of 4 spaces for indentation, in keeping with the rest of the repo. Also, instead of `@assert`ing conditions about arguments, you should use something like
```julia
has_offset_axes(a) && throw(ArgumentError("..."))
```

Reply:
thanks, fixed